### PR TITLE
Learning rate scheduler

### PR DESCRIPTION
**Pull request by terrytangyuan**

### CODE REVIEW

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  var count: Int = 0
+  private val logger = LoggerFactory.getLogger(classOf[FactorScheduler])
+
+  if (step < 1) {
+    throw new IllegalArgumentException("Schedule step must be greater or equal than 1 round")
+  }
+  if (factor >= 1.0) {
+    throw new IllegalArgumentException("Factor must be less than 1 to make lr reduce")
+  }
```
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala]: require(step < 1, "Schedule step must be greater or equal than 1 round") would be better, it throws IllegalArgumentException as well.

Comment (terrytangyuan) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala]: Thanks! Updated.

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package ml.dmlc.mxnet
+import org.slf4j.LoggerFactory
+
+/**
+ * Learning rate scheduler, which adaptively changes the learning rate
+ * based on the training progress.
+ * @author Yuan Tang
+ */
+
+abstract class LRScheduler(var baseLR: Double = 0.01) {
```
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala]: We should use Float wherever connected to NDArray
and 'protected' shall be enough

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  def apply(numUpdate: Int): Double = {
+
+    if (numUpdate > this.count + this.step) {
+      this.count += this.step
+      this.baseLR *= this.factor
+      this.logger.info(s"""Update$numUpdate: Change learning rate to ${this.baseLR}""")
+    }
+    this.baseLR
+  }
+}
```
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/LRScheduler.scala]: too many quotation marks. s"..." is enough

**Pull request merged**